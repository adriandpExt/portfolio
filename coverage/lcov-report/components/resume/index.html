
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/resume</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> components/resume</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.29% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>28/41</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/7</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">66.66% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>26/39</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Resume.tsx"><a href="Resume.tsx.html">Resume.tsx</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="73.33" class="pct medium">73.33%</td>
	<td data-value="15" class="abs medium">11/15</td>
	</tr>

<tr>
	<td class="file medium" data-value="TimelineCard.tsx"><a href="TimelineCard.tsx.html">TimelineCard.tsx</a></td>
	<td data-value="59.09" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 59%"></div><div class="cover-empty" style="width: 41%"></div></div>
	</td>
	<td data-value="59.09" class="pct medium">59.09%</td>
	<td data-value="22" class="abs medium">13/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="utils.tsx"><a href="utils.tsx.html">utils.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-06-22T07:57:52.871Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    