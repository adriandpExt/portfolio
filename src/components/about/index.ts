export * from "./About";
export * from "./Wrapper";
