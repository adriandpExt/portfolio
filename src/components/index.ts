export * from "./appbar";
export * from "./svg-icons";
export * from "./aside";
