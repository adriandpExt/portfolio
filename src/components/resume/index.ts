export * from "./Resume";
export * from "./TimelineCard";
