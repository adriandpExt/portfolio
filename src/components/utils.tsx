export const toggleDarkTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-900 text-white" : "bg-gray-200 text-gray-800";
};

export const toggleLightTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200 text-gray-800" : "bg-gray-900 text-white";
};

export const handleToggleDropIcon = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200  text-gray-800" : "bg-gray-900  text-white";
};
