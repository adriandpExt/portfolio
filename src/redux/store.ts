import { configureStore } from "@reduxjs/toolkit";
import linkReducers from "./reducer/linkReducers";
import contactReducers from "./reducer/contactReducers";

export const store = configureStore({
  reducer: {
    link: linkReducers,
    contact:contactReducers
  },
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
