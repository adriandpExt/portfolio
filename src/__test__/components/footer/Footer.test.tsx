import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";

import { Footer } from "../../../components";

import store from "../../../redux/store";

jest.mock("react-router-dom");

const footerScreen = (
  <Provider store={store}>
    <Footer />
  </Provider>
);

window.open = jest.fn();

describe("Footer component test", () => {
  test("should render footer ", () => {
    const { container } = render(footerScreen);

    expect(container).toMatchSnapshot();
  });

  test("linkedin open in new window", () => {
    const { getByTestId } = render(footerScreen);

    fireEvent.click(getByTestId("linkedin"));

    expect(window.open).toHaveBeenCalledWith(
      "https://www.linkedin.com/in/adrian-del-prado-285aa81b8",
      "_blank"
    );
  });

  test("facebook open in new window", () => {
    const { getByTestId } = render(footerScreen);

    fireEvent.click(getByTestId("facebook"));

    expect(window.open).toHaveBeenCalledWith(
      "https://www.facebook.com/adrian.delprado.98",
      "_blank"
    );
  });
});
