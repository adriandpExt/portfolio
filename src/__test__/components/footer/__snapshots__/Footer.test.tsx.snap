// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test should render footer  1`] = `
<div>
  <div
    class="MuiBottomNavigation-root css-ltaely-MuiBottomNavigation-root"
  >
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
      data-testid="facebook"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FacebookIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
      data-testid="linkedin"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LinkedInIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root MuiBottomNavigationAction-iconOnly css-3yflfb-MuiButtonBase-root-MuiBottomNavigationAction-root"
      data-testid="contact"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="EmailIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
        />
      </svg>
      <span
        class="MuiBottomNavigationAction-label MuiBottomNavigationAction-iconOnly css-1gh6k7f-MuiBottomNavigationAction-label"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
